app/serializers/user.js

```diff
@@ -5,7 +5,11 @@ export default ApplicationSerializer.extend({
   serialize(snapshot, options) {
     const json = this._super(...arguments);
     if (options && options.includeId) {
-      json.data.attributes = pick(json.data.attributes, ['email', 'password']);
+      if (snapshot.adapterOptions && snapshot.adapterOptions.updateMode === 'profile') {
```

Comment:
Change this such that only for register we send email and password. For anything else send all other attributes except email and password.